src/widget.tsx

```diff
@@ -0,0 +1,424 @@
+    super();
+    let layout = (this.layout = new PanelLayout());
+    this._listing = new Widget();
+    this._input = new URLInput();
+    layout.addWidget(this._input);
+    layout.addWidget(this._listing);
+    this.addClass('dask-DaskDashboardLauncher');
+    this._commands = options.commands;
+    this._items = options.items || DaskDashboardLauncher.DEFAULT_ITEMS;
+    this._input.urlChanged.connect(
+      this.update,
+      this
+    );
+  }
+
+  /**
+   * The list of dashboard items which can be launched.
+   */
+  get items(): DaskDashboardLauncher.IItem[] {
+    return this._items;
+  }
+
+  /**
+   * Get the URL input widget.
+   */
+  get input(): URLInput {
+    return this._input;
+  }
+
+  /**
+   * Handle an update request.
+   */
+  protected onUpdateRequest(msg: Message): void {
+    // Don't bother if the TOC is not visible
+    if (!this.isVisible) {
+      return;
+    }
+
+    ReactDOM.render(
+      <DashboardListing
+        isEnabled={this.input.isValid}
+        commands={this._commands}
+        items={this._items}
+      />,
+      this._listing.node
+    );
+  }
+
+  /**
+   * Rerender after showing.
+   */
+  protected onAfterShow(msg: Message): void {
+    this.update();
+  }
+
+  private _listing: Widget;
+  private _input: URLInput;
+  private _commands: CommandRegistry;
+  private _items: DaskDashboardLauncher.IItem[];
+}
+
+/**
+ * A widget for hosting a url input element.
+ */
+export class URLInput extends Widget {
+  /**
+   * Construct a new input element.
+   */
+  constructor() {
+    super();
+    this.addClass('dask-URLInput');
+    const wrapper = document.createElement('div');
+    wrapper.className = 'dask-URLInput-wrapper';
+    this._input = document.createElement('input');
+    this._input.placeholder = 'DASK DASHBOARD URL';
+    this.node.appendChild(wrapper);
+    wrapper.appendChild(this._input);
+
+    this._startUrlCheckTimer();
+  }
+
+  /**
+   * The underlying input value.
+   */
+  get input(): HTMLInputElement {
+    return this._input;
+  }
+
+  /**
+   * The base url for the dask webserver.
+   *
+   * #### Notes
+   * Setting this value will result in a urlChanged
+   * signal being emitted, but it will happen asynchronously,
+   * as it first checks to see whether the url is pointing
+   * at a valid dask dashboard server.
+   */
+  get url(): string {
+    return this._url;
+  }
+  set url(newValue: string) {
+    const oldValue = this._url;
+    if (newValue === oldValue) {
+      return;
+    }
+    Private.testDaskDashboard(newValue).then(result => {
+      this._url = newValue;
+      this._isValid = result;
+      this._input.value = newValue;
+      this._urlChanged.emit({ isValid: result, oldValue, newValue });
+      this._input.blur();
+      this.update();
+      if (!result) {
+        showErrorMessage(
+          'Invalid Dask URL',
+          Error(
+            `${newValue ||
+              'This'} does not appear to host a valid Dask dashboard`
```

Comment:
This still triggers when you open JLab and you have panes open from your last session but without any scheduler running yet.